### DISCUSSION TITLE
Issues extending get_client_token function

### DISCUSSION THREAD

**Discussion by Liana64** [saleor/saleor#8856] [Q&A]
I'm working with Saleor 2.11 and trying to implement a new payment gateway. The problem is that the payment processor requires a checkout amount and shipping details to create a client token. I can't find a way to pass this data to the get_client_token function without extending the plugin system itself. Any ideas, or am I out of luck here?

**Comment by karolkielecki** [comment]
@nyneava Hi! I've passed your issue to my colleague. There is currently a holiday season. Please expect a delay in our response. But first of all, we recommend using Saleor 3.0 beta. Saleor 3.0 introduces new features, including multichannel, and is much improved compared to 2.11.

**Reply by Liana64** [reply]
Thank you. I'm working on a solution to preserve the plugin manager by passing useCheckout data to a webhook but not sure if that's a great idea in production. I'll update the post if it works.
